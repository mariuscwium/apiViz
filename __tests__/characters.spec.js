import React from 'react';
import Characters from '../app/components/explorer/characters.jsx';
import renderer from 'react-test-renderer';

describe('Testing Characters component', () => {
    beforeEach(() => {
        fetch.resetMocks();
    });

    test('Displays when mounted', async () => {
        fetch.mockResponseOnce(JSON.stringify({ name: 'Luke Skywalker', url: 'test' }));

        const selectedFilm = {
            characters: [test]
        };

        let promise;
        const addPromise = newPromise => {
            promise = newPromise;
        };

        const component = renderer.create(
            <Characters selectedFilm={selectedFilm} selectedCharacter={false} addPromise={addPromise}>
                {state => {
                    if (state.data) {
                        return <div>{state.data.name}</div>;
                    } else {
                        return null;
                    }
                }}
            </Characters>
        );
        await promise;
        const instance = component.getInstance();
        instance.setState({ loading: false });

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
