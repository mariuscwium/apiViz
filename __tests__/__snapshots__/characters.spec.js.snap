// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Characters component Displays when mounted 1`] = `
<div
  style={
    Object {
      "position": "relative",
    }
  }
>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Select a character
  </h1>
  <ul
    className="sc-bdVaJa jJjsEG"
  >
    <li
      className="sc-bwzfXH fQvBvU"
      onClick={[Function]}
    >
      Luke Skywalker
    </li>
  </ul>
</div>
`;
